astropy/convolution/core.py

```diff
@@ -189,9 +189,9 @@ class Kernel1D(Kernel):
     ----------
     model : `~astropy.modeling.FittableModel`
         Model to be evaluated.
-    x_size : odd int, optional
-        Size of the kernel array. Default = 8 * width.
-    array : `~numpy.ndarray`
+    x_size : int, optional
+        Size of the kernel array. Default = 8 * width. Must be odd.
```

Comment:
Not your problem, but the default is not odd as written...

Reply:
Odd. I'll remove the note about being odd.

Reply:
✅

Reply:
Are you sure? I figured the width statement was wrong...

Reply:
Now checking the code, it does
```
_round_up_to_odd_integer(8 * stddev)
```

Reply:
It looks like this should be removed everywhere.

Reply:
8 times in convolution/kernels

Reply:
Looking through the code:
- Kernel doesn't have the argument, so it's fine.
- Kernel1D works on even or odd kernels but the default kernel should be changed to ⌊8*width+1⌋
- Kernel2D also works on even or odd kernels and likewise should be ⌊8*width+1⌋
- Gaussian1DKernel passes straight to Kernel1D
- Gaussian2DKernel should be ⌊8*stddev + 1⌋
- Box1DKernel  passes straight to Kernel1D
- etc.

So all arguments can be even or odd since Kernel1/2D can accept even or odd. I've changed the docstrings to  show the floor operator.